### PR TITLE
Support proxy for Gradle deployment

### PR DESCRIPTION
**Pull request by AlexeiVainshtein**

### CODE REVIEW

File: jfrog-cli/artifactory/utils/buildinfoproperties.go [hunk 1]
```diff
@@ -193,6 +207,25 @@ func CreateBuildInfoPropertiesFile(buildName, buildNumber string, config *viper.
 	return propertiesFile.Name(), nil
 }
 
+// If HTTP_PROXY env variable presented, adds to the config to deploy Gradle/Maven artifacts via proxy
+func setProxyIfDefined(config *viper.Viper) error {
+	// Add HTTP_PROXY if exists
+	proxy := os.Getenv(HttpProxy)
+	if proxy != "" {
+		url, err := url.Parse(proxy)
+		if err != nil {
```
Comment (eyalbe4) [inline jfrog-cli/artifactory/utils/buildinfoproperties.go]: CheckError missing.

File: jfrog-cli/artifactory/utils/buildinfoproperties.go [hunk 2]
```diff
@@ -193,6 +207,25 @@ func CreateBuildInfoPropertiesFile(buildName, buildNumber string, config *viper.
...
+			return err
+		}
+		host, port, err := net.SplitHostPort(url.Host)
```
Comment (eyalbe4) [inline jfrog-cli/artifactory/utils/buildinfoproperties.go]: CheckError missing.

File: jfrog-cli/artifactory/utils/buildinfoproperties.go [hunk 3]
```diff
@@ -193,6 +207,25 @@ func CreateBuildInfoPropertiesFile(buildName, buildNumber string, config *viper.
 	return propertiesFile.Name(), nil
 }
 
+// If HTTP_PROXY env variable presented, adds to the config to deploy Gradle/Maven artifacts via proxy
```
Comment (eyalbe4) [inline jfrog-cli/artifactory/utils/buildinfoproperties.go]: I suggest rephrasing the comment to:
// If the HTTP_PROXY environment variable is set, add to the config proxy details.

File: jfrog-cli/artifactory/utils/buildinfoproperties.go [hunk 4]
```diff
@@ -193,6 +207,25 @@ func CreateBuildInfoPropertiesFile(buildName, buildNumber string, config *viper.
...
+func setProxyIfDefined(config *viper.Viper) error {
```
Comment (eyalbe4) [inline jfrog-cli/artifactory/utils/buildinfoproperties.go]: In addition to the HTTP_PROXY env var, we should probably also support the *HTTPS_PROXY* env var.
See the *TestArtifactoryProxy* as a reference.

File: jfrog-cli/artifactory/utils/buildinfoproperties_test.go [hunk 5]
```diff
@@ -37,19 +57,30 @@ func testCreateDefaultPropertiesFile(buildType BuildType, t *testing.T) {
 		}
 	}
 
+	for key, value := range propertiesFileConfig {
```
Comment (eyalbe4) [inline jfrog-cli/artifactory/utils/buildinfoproperties_test.go]: Question - why do we need both the * propertiesFileConfig* and *expectedConfig* maps? Aren't they always identical?

File: jfrog-cli/artifactory/utils/buildinfoproperties_test.go [hunk 6]
```diff
@@ -28,6 +38,16 @@ func testCreateDefaultPropertiesFile(buildType BuildType, t *testing.T) {
 		t.Error(err)
 	}
 
+	var yamlConfig = map[string]string{
```
Comment (eyalbe4) [inline jfrog-cli/artifactory/utils/buildinfoproperties_test.go]: Is the *yamlConfig* map necessary?

File: jfrog-cli/artifactory/utils/buildinfoproperties_test.go [hunk 7]
```diff
@@ -134,3 +166,32 @@ func compareViperConfigs(t *testing.T, actual, expected *viper.Viper, buildType
 		}
 	}
 }
+
+func TestSetProxyIfNeeded(t *testing.T) {
```
Comment (eyalbe4) [inline jfrog-cli/artifactory/utils/buildinfoproperties_test.go]: Question: Isn't this test sufficient for testing the proxy functionality? If it is, can we remove the new additions to the other tests in this file?

File: jfrog-cli/artifactory/utils/buildinfoproperties.go (hunk 4 above)
Comment (AlexeiVainshtein) [reply jfrog-cli/artifactory/utils/buildinfoproperties.go]: We don't support https in the build info. We only support http. I've checked and the proxy is usually running on http only without https.

File: jfrog-cli/artifactory/utils/buildinfoproperties_test.go (hunk 7 above)
Comment (AlexeiVainshtein) [reply jfrog-cli/artifactory/utils/buildinfoproperties_test.go]: I think that this only test the setProxyIfDefined function. The other tests testing the entire viper that the information is written correctly in the form of artifactory.proxy.host and artifactory.proxy.port. Those are the values that the build info needs so the other tests purpose is to check that those are the correct keys.

File: jfrog-cli/artifactory/utils/buildinfoproperties_test.go (hunk 6 above)
Comment (AlexeiVainshtein) [reply jfrog-cli/artifactory/utils/buildinfoproperties_test.go]: I removed it

File: jfrog-cli/artifactory/utils/buildinfoproperties_test.go (hunk 5 above)
Comment (AlexeiVainshtein) [reply jfrog-cli/artifactory/utils/buildinfoproperties_test.go]: Not exactly. The expectedConfig is a Viper not a map. After creating the viper and populating the viper information we adding the propertiesFileConfig key and value to the viper. We can change this to:

	expectedConfig.Set("artifactory.proxy.host", host)
	expectedConfig.Set("artifactory.proxy.port", port)

File: jfrog-cli/artifactory/utils/buildinfoproperties.go (hunk 1 above)
Comment (AlexeiVainshtein) [reply jfrog-cli/artifactory/utils/buildinfoproperties.go]: Done

File: jfrog-cli/artifactory/utils/buildinfoproperties.go (hunk 2 above)
Comment (AlexeiVainshtein) [reply jfrog-cli/artifactory/utils/buildinfoproperties.go]: Done

File: jfrog-cli/artifactory/utils/buildinfoproperties.go (hunk 3 above)
Comment (AlexeiVainshtein) [reply jfrog-cli/artifactory/utils/buildinfoproperties.go]: Done

File: jfrog-cli/artifactory/utils/buildinfoproperties_test.go (hunk 5 above)
Comment (AlexeiVainshtein) [reply jfrog-cli/artifactory/utils/buildinfoproperties_test.go]: I've changed it per our conversation.

**Pull request merged**